node/npm_dependencies.ts

```diff
@@ -72,26 +124,39 @@ const getNPMSpecifiers = async (basePath: string, functions: string[], importMap
       return nftResolve(specifier, ...args)
     },
   })
-  const npmSpecifiers = new Set<string>()
+  const npmSpecifiers: { specifier: string; types?: string }[] = []
   const npmSpecifiersWithExtraneousFiles = new Set<string>()
 
-  reasons.forEach((reason, filePath) => {
-    const packageName = getPackageName(filePath)
+  for (const [packageJsonPath, reason] of reasons.entries()) {
+    // every dependency will have its `package.json` in `reasons` exactly once.
+    // by only looking at this file, we save us from doing duplicate work.
+    const isPackageJson = packageJsonPath.endsWith('package.json')
+    if (!isPackageJson) continue
```

Comment:
I think this means we won't detect extraneous files? I think we can only discard a file that isn't `package.json` for the purpose of detecting direct dependencies, finding types, etc.

Reply:
Good point, yeah! I was assuming we have a test for that, but apparently we don't. Gonna add that later today.

Reply:
addressed in https://github.com/netlify/edge-bundler/pull/505/commits/e206df474a2ae856b5c7d5bdfb7d1a96eac2f480